Importing libraries

In [ ]:
import os
import pandas as pd
import numpy as np
import matplotlib as plt
from sklearn.model_selection import train_test_split
from sklearn import preprocessing


Importing Dataset

To enable the code to access the housing dataset, please upload the "housing.csv" file to the runtime, and subsequently, run the provided code.

Important: I am including the "housing.csv" dataset file along with my assignment submission on Canvas.


In [ ]:
df = pd.read_csv('housing.csv')
df.isnull().sum()


Index(['longitude', 'latitude', 'housing_median_age', 'total_rooms',
       'total_bedrooms', 'population', 'households', 'median_income',
       'median_house_value'],
      dtype='object')

Dropping null values and irrelevant columns

In [ ]:
df.dropna(inplace=True)
df=df.drop('ocean_proximity', axis=1)
df.columns

Traing Test splitting the dataset

In [ ]:
train,test = train_test_split(df,test_size=0.20,random_state=42)

In [ ]:
train.head()

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
17727    -121.80     37.32                14.0       4412.0           924.0   
2057     -119.63     36.64                33.0       1036.0           181.0   
6453     -118.06     34.12                25.0       3891.0           848.0   
4619     -118.31     34.07                28.0       2362.0           949.0   
15266    -117.27     33.04                27.0       1839.0           392.0   

       population  households  median_income  median_house_value  
17727      2698.0       891.0         4.7027            227600.0  
2057        620.0       174.0         3.4107            110400.0  
6453       1848.0       759.0         3.6639            248100.0  
4619       2759.0       894.0         2.2364            305600.0  
15266      1302.0       404.0         3.5500            214600.0

X_train, X_test, Y_train, Y_test

In [ ]:
x_train = train.drop('median_house_value',axis=1)
x_test = test.drop('median_house_value',axis=1)

In [ ]:
y_train = train.median_house_value
y_test = test.median_house_value

In [ ]:
x_train.head()

longitude  latitude  housing_median_age  total_rooms  total_bedrooms  \
17727    -121.80     37.32                14.0       4412.0           924.0   
2057     -119.63     36.64                33.0       1036.0           181.0   
6453     -118.06     34.12                25.0       3891.0           848.0   
4619     -118.31     34.07                28.0       2362.0           949.0   
15266    -117.27     33.04                27.0       1839.0           392.0   

       population  households  median_income  
17727      2698.0       891.0         4.7027  
2057        620.0       174.0         3.4107  
6453       1848.0       759.0         3.6639  
4619       2759.0       894.0         2.2364  
15266      1302.0       404.0         3.5500

In [ ]:
y_train.head()

17727    227600.0
2057     110400.0
6453     248100.0
4619     305600.0
15266    214600.0
Name: median_house_value, dtype: float64

Linear Regression Model

In [ ]:
from sklearn.linear_model import LinearRegression
model = LinearRegression()
model.fit(x_train,y_train)



LinearRegression()

Model Evaluation

In [ ]:
#Training accuracy
model.score(x_train,y_train)

0.6360185727313741

In [ ]:
predictions = model.predict(x_test)

In [ ]:
#Testing accuracy
model.score(x_test,y_test)

0.640086568899372


Given that the training accuracy of our linear regression model is 63.6%, and the testing accuracy is 64%, I aim to enhance the model's predictive performance through hyperparameter tuning. Consequently, I have introduced Random Forest Regression to explore potential improvements.

In [ ]:
from sklearn.ensemble import RandomForestRegressor

Random Forest Regressor Model

In [ ]:
my_new_model = RandomForestRegressor()

In [ ]:
my_new_model.fit(x_train,y_train)

RandomForestRegressor()

Evaluarion of model

In [ ]:
#Training accuracy
my_new_model.score(x_train,y_train)

0.9748999844488277

In [ ]:
output = my_new_model.predict(x_test)

In [ ]:
#Testing accuracy
my_new_model.score(x_test,y_test)

0.8232306240796442

The application of Random Forest Regression has resulted in a notable enhancement in testing accuracy, escalating it from 63% to 82%.